tests/phpunit/tests/rest-api/rest-server.php

```diff
@@ -77,6 +77,62 @@ public function test_envelope() {
 		$this->assertSame( $headers, $enveloped['headers'] );
 	}
 
+	/**
+	 * @dataProvider data_embed_params
+	 * @ticket 54015
+	 */
+	public function test_envelope_param( $_embed ) {
+		// Register our testing route.
+		rest_get_server()->register_route(
+			'test',
+			'/test/embeddable',
+			array(
+				'methods'  => 'GET',
+				'callback' => array( $this, 'embedded_response_callback' ),
+			)
+		);
+		$data    = array(
+			'amount of arbitrary data' => 'alot',
+		);
+		$status  = 987;
+		$headers = array(
+			'Arbitrary-Header' => 'value',
+			'Multiple'         => 'maybe, yes',
+		);
+
+		$response = new WP_REST_Response( $data, $status );
+		$response->header( 'Arbitrary-Header', 'value' );
+
+		// Check header concatenation as well.
+		$response->header( 'Multiple', 'maybe' );
+		$response->header( 'Multiple', 'yes', false );
+
+		// All others should be embedded.
+		$response->add_link( 'alternate', rest_url( '/test/embeddable' ), array( 'embeddable' => true ) );
+
+		$embed             = rest_parse_embed_param( $_embed );
+		$envelope_response = rest_get_server()->envelope_response( $response, $embed );
+
+		// The envelope should still be a response, but with defaults.
+		$this->assertInstanceOf( 'WP_REST_Response', $envelope_response );
```

Comment:
```suggestion
		$this->assertInstanceOf( WP_REST_Response::class, $envelope_response );
```

Kinda personal preference, but we've been doing this with class instances recently. Let's us have IDE completion and not use a stringly typed API.